python/gtsam/examples/SFMExample_bal.py

```diff
@@ -60,59 +69,61 @@ def run(args):
 
     # Add a prior on pose x1. This indirectly specifies where the origin is.
     graph.push_back(
-        gtsam.PriorFactorPinholeCameraCal3Bundler(
-            C(0), scene_data.camera(0), gtsam.noiseModel.Isotropic.Sigma(9, 0.1)
-        )
-    )
+        PriorFactorPinholeCameraCal3Bundler(
+            C(0), scene_data.camera(0),
+            gtsam.noiseModel.Isotropic.Sigma(9, 0.1)))
     # Also add a prior on the position of the first landmark to fix the scale
     graph.push_back(
-        gtsam.PriorFactorPoint3(
-            P(0), scene_data.track(0).point3(), gtsam.noiseModel.Isotropic.Sigma(3, 0.1)
-        )
-    )
+        PriorFactorPoint3(P(0),
+                          scene_data.track(0).point3(),
+                          gtsam.noiseModel.Isotropic.Sigma(3, 0.1)))
 
     # Create initial estimate
     initial = gtsam.Values()
-    
+
     i = 0
     # add each PinholeCameraCal3Bundler
     for cam_idx in range(scene_data.number_cameras()):
         camera = scene_data.camera(cam_idx)
         initial.insert(C(i), camera)
         i += 1
 
-    j = 0
     # add each SfmTrack
     for t_idx in range(scene_data.number_tracks()):
-        track = scene_data.track(t_idx)  
-        initial.insert(P(j), track.point3())
-        j += 1
+        track = scene_data.track(t_idx)
+        initial.insert(P(t_idx), track.point3())
 
     # Optimize the graph and print results
     try:
         params = gtsam.LevenbergMarquardtParams()
         params.setVerbosityLM("ERROR")
         lm = gtsam.LevenbergMarquardtOptimizer(graph, initial, params)
         result = lm.optimize()
-    except Exception as e:
+    except RuntimeError:
         logging.exception("LM Optimization failed")
         return
+
     # Error drops from ~2764.22 to ~0.046
-    logging.info(f"final error: {graph.error(result)}")
+    logging.info("initial error: %f", graph.error(initial))
+    logging.info("final error: %f", graph.error(result))
 
+    plot(scene_data, result)
 
-if __name__ == "__main__":
+
+def main():
```

Comment:
Adding a return type declaration for None is questionable, since is it even useful? If the return type is something non-trivial, then it helps with things like linting and autocomplete, but None doesn't help with any of that.

Reply:
I think it's useful purely for complete static analysis

Reply:
But does the static analysis report it missing? I have all warning enabled (even for the linter) and I get no warnings.